include/nuttx/spinlock.h

```diff
@@ -278,8 +278,6 @@ spin_lock_nopreempt(FAR volatile spinlock_t *lock)
 #ifdef CONFIG_SPINLOCK
 static inline_function void spin_lock(FAR volatile spinlock_t *lock)
 {
-  sched_lock();
```

Comment:
But spinning locks implicitly disable preemption:
https://kernel.googlesource.com/pub/scm/linux/kernel/git/christoph/linux/+/refs/heads/arm64_maxsmp/Documentation/locking/locktypes.rst#76

Reply:
please use **spin_lock_nopreempt()** , this PR just restore the code behavior to the same as before